### DISCUSSION TITLE
Why redefinition of MAX_ROUTING_LAYER is not possible for most platforms?

### DISCUSSION THREAD

**Discussion by jtremesay** [The-OpenROAD-Project/OpenROAD-flow-scripts#1469] [Q&A] [answered]
Hi,

I'm trying to create a hierarchical  flow for a SoC design with nangate45.
For the sub-designs, I wanted to limit the routing to metal 6.

From what I see in the [documentation](https://openroad-flow-scripts.readthedocs.io/en/latest/user/FlowVariables.html#routing), `MAX_ROUTING_LAYER` is the variable I should use.

But the documentation also says that the variable is already set for nangate45 and cannot be reassigned.

what is the reason for this choice?

Can I safely edit this [line](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/8e36a089b73820e55312e0f7e9b5e20387133d53/flow/platforms/nangate45/config.mk#L88) to `export MAX_ROUTING_LAYER ?= metal10` ?

**Comment by maliberty** [comment, marked as answer]
I think that's fine.  You can make a PR for the change if you wish.

**Comment by jtremesay** [comment]
I'm going to do some local tests and make a PR if thinks look ok

**Comment by maliberty** [comment]
Note that you may also need to look at PDN and the layers it uses for your sub-block.

**Comment by jtremesay** [comment]
PR available [here](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/pull/1470)
